nomad/job_endpoint_hook_implicit_identities.go

```diff
@@ -19,33 +19,57 @@ func (jobImplicitIdentitiesHook) Name() string {
 
 func (h jobImplicitIdentitiesHook) Mutate(job *structs.Job) (*structs.Job, []error, error) {
 	for _, tg := range job.TaskGroups {
+		var hasIdentity bool
+
 		for _, s := range tg.Services {
-			h.handleConsulService(s)
+			if h.handleConsulService(s) {
+				hasIdentity = true
+			}
 		}
 
 		for _, t := range tg.Tasks {
 			for _, s := range t.Services {
-				h.handleConsulService(s)
+				if h.handleConsulService(s) {
+					hasIdentity = true
+				}
 			}
 			if len(t.Templates) > 0 {
-				h.handleConsulTasks(t)
+				if h.handleConsulTasks(t) {
+					hasIdentity = true
+				}
 			}
-			h.handleVault(t)
+			if h.handleVault(t) {
+				hasIdentity = true
+			}
+		}
```

Comment:
Instead of checking for identities in each method can we just do a `hasIdentity = hasIdentity || (len(t.Identities) > 0)` here?

Reply:
We'd also have to check the services, but yeah that seems nicer.